### PR TITLE
feat(minifier): minimize computed property access

### PR DESCRIPTION
**Pull request by sapphi-red**
Ported `ConvertToDottedProperties` partially. The rest seems to be mostly handled by #8147.

### CODE REVIEW

File: crates/oxc_minifier/src/ast_passes/convert_to_dotted_properties.rs [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        test_same("a[':']");
+        test_same("a['.']");
+        test_same("a['0']");
+        test_same("a['p ']");
+        test_same("a['p' + '']");
+        test_same("a[p]");
+        test_same("a[P]");
+        test_same("a[$]");
+        test_same("a[p()]");
+        // test_same("a['default']");
```
Comment (sapphi-red) [inline crates/oxc_minifier/src/ast_passes/convert_to_dotted_properties.rs:106]: Keywords seems to be not allowed for ES3, but I guess this is fine for us.

File: crates/oxc_minifier/src/ast_passes/convert_to_dotted_properties.rs [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        // Ignorable control characters are ok in Java identifiers, but not in JS.
+        test_same("a['A\\u0004']");
+        // upper case lower half of o from phonetic extensions set.
+        // valid in Safari, not in Firefox, IE.
+        // test_same("a['\\u1d17A']");
+        // Latin capital N with tilde - nice if we handled it, but for now let's
+        // only allow simple Latin (aka ASCII) to be converted.
+        // test_same("a['\\u00d1StuffAfter']");
```
Comment (sapphi-red) [inline crates/oxc_minifier/src/ast_passes/convert_to_dotted_properties.rs:114]: `\u1d17A` is invalid for ES5 (ES5 uses Unicode 3.0 and that character was introduced in Unicode 3.1) and valid for ES2015 (ES2015 uses Unicode 5.1).
`\u00d1` is valid for ES3+.

**Comment by Boshen** [comment]
This falls under substitute alternative syntax. I'll move the code around when I get the time.

**Comment by sapphi-red** [comment]
=> #8175, #8176

**Pull request closed without merging**